### PR TITLE
Bungeecord plugin messaging channel, broadcasting

### PR DESCRIPTION
**Pull request by maol3**
Added subchannel Alert for the Bungeecord plugin messaging channel

### CODE REVIEW

File: proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java [hunk 1]
```diff
@@ -437,6 +437,14 @@ public void handle(PluginMessage pluginMessage) throws Exception
                     player.disconnect( new TextComponent( kickReason ) );
                 }
             }
+            if ( subChannel.equals( "Alert" ) )
+            {
+                String message = in.readUTF();
+                if ( message != null && message != "" )
```
Comment (Janmm14) [inline proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java]: In Java strings should never be compared by identity, use `!message.isEmpty()` here and in general `equals()`.

**Comment by Mystiflow** [comment]
I think naming the subchannel "Broadcast" makes more sense.

**Comment by Janmm14** [comment]
The command for broadcasts on bungee is /alert, so I can understand both suggestions.

**Comment by md-5** [comment]
Should Message just be expanded to allow ALL (and I guess a json version added)

File: proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java [hunk 2]
```diff
@@ -391,10 +391,19 @@ public void handle(PluginMessage pluginMessage) throws Exception
             }
             if ( subChannel.equals( "Message" ) )
             {
-                ProxiedPlayer target = bungee.getPlayer( in.readUTF() );
-                if ( target != null )
+                String target = in.readUTF();
+                if ( target.equals("ALL") )
```
Comment (Janmm14) [inline proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java]: Suggesting to include a character illegal in player names

**Comment by md-5** [comment]
Thanks should be good now. Also this PR didn't compile. Definitely need to test these things :)

**Pull request closed without merging**